python_modules/libraries/dagster-dg/dagster_dg/cli/init.py

```diff
@@ -149,4 +179,40 @@ def init_command(
         python_environment=DgProjectPythonEnvironment.from_flag(python_environment),
     )
 
-    click.echo("You can create additional projects later by running 'dg scaffold project'.")
+    if workspace:
+        click.echo("You can create additional projects later by running 'dg scaffold project'.")
+
+    venv_path = project_path / ".venv"
+
+    # This already will have occurred during the scaffolding step
+    if python_environment == "uv_managed":
+        run_uv_sync = False
+    # If a venv already exists, only run `--uv-sync` if it is explicitly set.
+    elif venv_path.exists():
+        run_uv_sync = uv_sync is True
+    else:
+        if uv_sync is True:
+            run_uv_sync = True
+        elif uv_sync is None and is_uv_installed():
+            response = click.prompt(
+                "Run `uv sync` to bootstrap a uv.lock and virtual environment (y/n)?", default="y"
+            )
+            run_uv_sync = response.lower() == "y"
+        else:
+            run_uv_sync = False
```

Comment:
I think this would be nicer in its own function so that we don't have to deal with `run_uv_sync` without a definition in the outer scope.